src/spring-mongo/src/main/java/net/conology/spring/restjsonpath/mongo/ir/MongoAnyOfSelector.java

```diff
@@ -0,0 +1,36 @@
+package net.conology.spring.restjsonpath.mongo.ir;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.springframework.data.mongodb.core.query.Criteria;
+
+public final class MongoAnyOfSelector implements MongoAlternativesSelector, MongoSelector {
```

Comment:
AllOf is not the same as end, which is why the other is called allOf. Here we should definitely use Or. Don#t introduce to many names for the same concepts